File: c461(logic_operators)/Python/logic_operators.py
from sys import stdin, stdout
while True:
    line = stdin.readline()
    if not line: break
    a, b, r = map(lambda x: bool(int(x)), line.split())
    p = True
    if (a & b) == r:
        p = False
        stdout.write('AND\n')
    if (a | b) == r:
        p = False
        stdout.write('OR\n')
    if (a ^ b) == r:
        p = False
        stdout.write('XOR\n')
    if p:
        stdout.write('IMPOSSIBLE\n')

File: c461(logic_operators)/C++/logic_operators.cpp
#include <iostream>

int main(void) {
  int a, b, r;
  while (std::cin >> a >> b >> r) {
    a = a && 1;
    b = b && 1;
    bool p = true;
    if ((a & b) == r) {
      p = false;
      std::cout << "AND\n";
    }
    if ((a | b) == r) {
      p = false;
      std::cout << "OR\n";
    }
    if ((a ^ b) == r) {
      p = false;
      std::cout << "XOR\n";
    }
    if (p) std::cout << "IMPOSSIBLE\n";
  }
  return 0;
}
